pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -293,6 +294,24 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	// the attached volumes field. This function ensures that we sync with
 	// the actual status.
 	adc.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+
+	if _, exists := node.Annotations[volumehelper.ControllerManagedAttachAnnotation]; exists {
+		// Newly added node is managed by controller, ensure all pods are up to date.
+		// Run in a separate goroutine so we don't block the node handler goroutine.
+		go adc.processPodsForNode(nodeName)
+	}
+}
+
+func (adc *attachDetachController) processPodsForNode(nodeName types.NodeName) {
```

Comment:
rename to `fetchAndProcessPodsForNode(...)` -- I want the name to emphasis that it is not passively processing but actively fetching.

Also add a comment summarizing what this method does above it.